### ISSUE TITLE
generate downsample image is not unabridged picture

### ISSUE THREAD

**Issue by Dylan-Jinx** [JaehaKim97/Adaptive-Downsampling-Model#3]
I use `DIV2K x2` dataset to generate downsample picture,and `iter=80`,but the generate downsample image is not unabridged picture like this.
![0657](https://user-images.githubusercontent.com/57589991/219934703-c859dc52-555d-425d-baea-33649a906a02.png)

**Comment by JaehaKim97** [comment]
Do you have a training checkpoint in `../experiments/save_name/model/training_down_0080.pth`?

If it exists, then you can again make downsampled images through the model by the following command:

`CUDA_VISIBLE_DEVICES=0 python test_down.py --source DIV2K-HR --name make_down_again --resume_down ../experiments/save_name/model/training_down_0080.pth`

**Comment by JaehaKim97** [comment]
In addition, the results will be located in `../results/make_down_again`.

**Comment by Dylan-Jinx** [comment]
If I want to simulate real world downsampling with camera specific properties, how do I construct my training set, i.e. I take the pth file after training and run the above command to give me the images I want, and then simulate real world downsampling?

**Comment by JaehaKim97** [comment]
Yes, it is.

The generated images will be the images that simulate the downsampling of the Target image set.

**Comment by Dylan-Jinx** [comment]
I'm very recently busy until now to look up your advice,I used your advice to rerun this model,and `dataset is div2k_valid_hd`.`weight file is training_down_0080.pth`,I found the image resolution degradation,and whole feeling of image is not remarkable change.I still think the picture is sharp, but the resolution has been lowered.My result like this
vaild reference HD image
![0801](https://user-images.githubusercontent.com/57589991/221110827-ef28e3ad-cb11-41c6-aaf2-8a112c1739c7.png)
the result LD image
![0801](https://user-images.githubusercontent.com/57589991/221110878-0105fd31-8e6f-48a0-95fd-ed6661da4935.png)

I don't know what else has changed besides the resolution,please help me solve this confusion.

**Comment by JaehaKim97** [comment]
I'm sorry for not getting back to you sooner.
To investigate whether the code was wrong, I reran my whole training code but **did not find any problems**.

Below, I have provided you with more detailed steps for reproducing. Please carefully check each step.

### 0. Environment
Ubuntu 18.04, RTX 2080 TI, CUDA 11.2

### 1. Dataset preparation
Please locate the DIV2K dataset in the ```datasets``` as the following format:

`├──DIV2K-HR-train`
     `├──0001.png`
     `...`
     `└──0400.png` <br/> 
`├──DIV2K-LR-train-bicx2`
     `├──0401x2.png`
     `...`
     `└──0800x2.png` <br/>
`└──DIV2K-HR-val`
     `├──0801.png`
     `...`
     `└──0900.png`

### 2. Training
```CUDA_VISIBLE_DEVICES=0 python train.py --source DIV2K-HR-train --target DIV2K-LR-train-bicx2 --name Double-check-DIV2K-LR-bicx2 --make_down```

As the training proceeds, the estimated downsampling kernel will be saved in ```experiments/Double-check-DIV2K-LR-bicx2/estimated_kernels/*```.

Since the target degradation is the bicubic downsampling, the estimated kernel results in 80 epochs will look as below:

![kernel_80](https://user-images.githubusercontent.com/41074450/221731591-a594c50b-8617-4570-aa5e-23bd32d07e87.png)

### 3. Validation
For validating the trained model on the DIV2K-validation set, please run the below command:

```CUDA_VISIBLE_DEVICES=0 python test_down.py --source DIV2K-HR-val --name From-DIV2K-HR-val-bicx2 --resume_down ../experiments/Double-check-DIV2K-LR-bicx2/models/training_down_0080.pt```

The downsampled results will be located in ```results/From-DIV2K-HR-val-bicx2/down_results/ep_001/*```.

#### 3-1) Quantitative
The reproduced PSNR between ground-truth bicubicx2 DIV2K validation set (801x2.png - 900x2.png) and downsampled results from the ADM is 45.41 (45.83 in the paper).

#### 3-2) Qualitative

HR (801.png)
![0801](https://user-images.githubusercontent.com/41074450/221732947-9d52f2b1-3f27-4030-94ce-8ad18eaa0fb3.png)

A downsampled result from our ADM
![0801](https://user-images.githubusercontent.com/41074450/221733018-5fbd95e6-8f90-4fe5-a3cf-436693dea036.png)

GT downsampled image (bicubicx2)
![0801x2](https://user-images.githubusercontent.com/41074450/221733149-5ae563a3-1597-402d-be57-b5d5b9b13271.png)

### 4. Additional Comment

Our ADM is designed to simulate the degradation of target images.

Therefore, if the target degraded images are not that blurry (i.e., small spatial kernel size), the generated downsampled images from ADM will also follow that property.

However, your attached images seem to undergo some color-shifting problems (as in yellow hair on the head), and I guess that there exists some problem in training.

Please follow the detailed introduction, and let me know if you obtain different results from the above.

**Comment by Dylan-Jinx** [comment]
@JaehaKim97 ,Heello,Now I am carefully following your above guide to conduct the experiment again, but I have another question, I have seen another Issue that is can imitate a particular camera style, so I need to train a sampling model with a particular camera style, then what should I do

**Comment by JaehaKim97** [comment]
What do you mean by the "sampling model"?

Our ADM learns a target-specific downsampling model.
Therefore, to learn the downsampling model for a particular camera style, you must re-train the ADM by setting the images taken from that camera as the target image set.

**Comment by Dylan-Jinx** [comment]
my problem like issue `Questions about the model `.For example, if I want to simulate the downsampling of Nikon D5300 camera, then my HD image is arbitrary, and then the LR image is the result taken by D5300? And if so, how many photos would be appropriate for training?

**Comment by JaehaKim97** [comment]
Yes, your understanding is correct. Please refer to [Dataset preparation](https://github.com/JaehaKim97/Adaptive-Downsampling-Model#-dataset-preperation).

**Comment by Dylan-Jinx** [comment]
But if I need to train a model for this particular camera, what is the right number of my LR images to have while keeping the variety high? I currently only have about 200-400 images of various types such as people, landscapes, landscapes and so on. I think this amount should be enough for LR data, right?

**Comment by JaehaKim97** [comment]
It should be enough. As I noted in [Dataset preparation](https://github.com/JaehaKim97/Adaptive-Downsampling-Model#-dataset-preperation), we recommend using more than 200 in HD scale.

**Comment by Dylan-Jinx** [comment]
I think I should know how to organize LR pictures for training now. Thank you very much for your guidance.
